### PR TITLE
Bugfix/220 fix linting

### PR DESCRIPTION
**Pull request by 0x6431346e**
Fixes: #220 

Tried to install `documentation` as well, but is complaining about us using `babel@6`.
I think we can add it later, after we upgrade to 7, to keep this PR smaller.

(I tested `init` and `run` commands with `npm link`)

### CODE REVIEW

**Comment by 0x6431346e** [comment]
Upgrading the dependencies did not work, this is the error `documentation lint` throws btw:
```
{ Error: Requires Babel "^7.0.0-0", but was loaded with "6.26.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel. while parsing file: /home/daniel/repos/aragon/aragon-cli/dist/cli.js
    at throwVersionError (/home/daniel/repos/aragon/aragon-cli/node_modules/@babel/helper-plugin-utils/lib/index.js:65:11)
    at Object.assertVersion (/home/daniel/repos/aragon/aragon-cli/node_modules/@babel/helper-plugin-utils/lib/index.js:13:11)
    at _default (/home/daniel/repos/aragon/aragon-cli/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js:41:7)
    at /home/daniel/repos/aragon/aragon-cli/node_modules/@babel/helper-plugin-utils/lib/index.js:19:12
    at Function.memoisePluginContainer (/home/daniel/repos/aragon/aragon-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:113:13)
    at Function.normalisePlugin (/home/daniel/repos/aragon/aragon-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:146:32)
    at /home/daniel/repos/aragon/aragon-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:184:30
    at Array.map (<anonymous>)
    at Function.normalisePlugins (/home/daniel/repos/aragon/aragon-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:158:20)
    at OptionManager.mergeOptions (/home/daniel/repos/aragon/aragon-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:234:36)
    at OptionManager.init (/home/daniel/repos/aragon/aragon-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
    at File.initOptions (/home/daniel/repos/aragon/aragon-cli/node_modules/babel-core/lib/transformation/file/index.js:212:65)
    at new File (/home/daniel/repos/aragon/aragon-cli/node_modules/babel-core/lib/transformation/file/index.js:135:24)
    at Pipeline.transform (/home/daniel/repos/aragon/aragon-cli/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at Babelify._flush (/home/daniel/repos/aragon/aragon-cli/node_modules/documentation/node_modules/babelify/index.js:26:24)
    at Babelify.prefinish (_stream_transform.js:137:10)
    at emitNone (events.js:106:13)
    at Babelify.emit (events.js:208:7)
    at prefinish (_stream_writable.js:603:14)
    at finishMaybe (_stream_writable.js:611:5)
    at endWritable (_stream_writable.js:622:3)
    at Babelify.Writable.end (_stream_writable.js:573:5)
    at DestroyableTransform.onend (/home/daniel/repos/aragon/aragon-cli/node_modules/readable-stream/lib/_stream_readable.js:577:10)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
  code: 'BABEL_VERSION_UNSUPPORTED',
  version: '6.26.3',
  range: '^7.0.0-0',
  filename: '/home/daniel/repos/aragon/aragon-cli/dist/cli.js' }

```

**Comment by sohkai** [comment]
@0x6431346e It might be the latest version of documentation. It looks like `documentation@9.0.0-alpha.0` is their breaking change for `babel@7`, so older versions should work. `aragon.js` is also (right now) on `babel@6` and using `documentation@8.1.2`.

**Comment by 0x6431346e** [comment]
This is weird, it did not work with `8.1.2` but with `9.0.0-alpha.0` :confused: 

I guess it makes sense because we use babel@7, except for the polyfill which is still at 6.

**Comment by sohkai** [comment]
Weird, we should definitely move the polyfill up to `babel@7`!

**Comment by 0x6431346e** [comment]
Upgraded babel and tested `aragon init` and `aragon run`, seems ok :ok_hand:

File: src/commands/dao_cmds/acl_cmds/utils/knownRoles.js [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
 // TODO: add support for user apps
 const rolesForApps = () => {
-	const allRoles = defaultAppsRoles.concat(currentAppRoles())
-	const knownRoles = allRoles.reduce(
-		(acc, role) => Object.assign(acc, { [keccak256(role.id)]: role })
-	, {}) 
+  const allRoles = defaultAppsRoles.concat(currentAppRoles())
+  const knownRoles = allRoles.reduce(
+    (acc, role) => Object.assign(acc, { [keccak256(role.id)]: role })
+    , {})
```
Comment (sohkai) [inline src/commands/dao_cmds/acl_cmds/utils/knownRoles.js]: Looks a bit weird, I think it should be:

```js
allRoles.reduce(
  (acc, role) => Object.assign(acc, { [keccak256(role.id)]: role }),
  {}
)
```

File: src/commands/apm_cmds/publish.js [hunk 2]
```diff
@@ -323,7 +322,7 @@ exports.task = function ({
     },
     {
       title: 'Building frontend',
-      enabled: () => build,
+      enabled: () => build && !http,
```
Comment (sohkai) [inline src/commands/apm_cmds/publish.js]: Nice find!

File: src/commands/apm_cmds/publish.js [hunk 3]
```diff
@@ -222,7 +221,7 @@ exports.task = function ({
               return `${module.version} is a valid version`
             }
 
-            throw new MessageError(module
+            throw new Error(module
```
Comment (sohkai) [inline src/commands/apm_cmds/publish.js]: Not sure about the `MessageError()`

File: src/helpers/ipfs-daemon.js [hunk 4]
```diff
@@ -45,7 +45,6 @@ const isIPFSCORS = async (ipfsRpc) => {
     ${IPFSCORS.map(({ key, value }) => {
       return `${key}: ${value}`
     }).join('\n    ')}`)
-    process.exit()
```
Comment (sohkai) [inline src/helpers/ipfs-daemon.js]: Just wanted to highlight this in case we needed it.

File: src/commands/dao_cmds/utils/aragonjs-wrapper.js [hunk 5]
```diff
@@ -51,20 +46,20 @@ const initWrapper = async (
...
-        registryAddress: ensRegistryAddress,
-      })
+      provider,
+      registryAddress: ensRegistryAddress
+    })
     : dao
 
   if (!daoAddress) {
-    onError(new InvalidAddress('The provided DAO address is invalid'))
+    onError(new Error('The provided DAO address is invalid'))
```
Comment (sohkai) [inline src/commands/dao_cmds/utils/aragonjs-wrapper.js]: Not sure if we need the `InvalidAddress` error

File: src/helpers/ipfs-daemon.js (hunk 4 above)
Comment (0x6431346e) [reply src/helpers/ipfs-daemon.js]: I've added it in the handler where this task is called, as we cannot do both throw and exit

**Pull request merged**